### PR TITLE
master

### PR DESCRIPTION
**Pull request by wamiikechukwu**

### CODE REVIEW

**Comment by alvinmarshall** [comment]
Pull request to master not develop. Develop is my branch. So you face merge conflict

**Comment by wamiikechukwu** [comment]
Sorry about that... i was trying to see how these things work.
I notices that changes are pushed to the master branch, even if the code fail testing

**Comment by wamiikechukwu** [comment]
i feel the rules for the branching is not observed

**Comment by alvinmarshall** [comment]
@wamiikechukwu  select the master branch and set the rules properly, i can't set those rules here

File: app/src/main/java/com/me/registrationtask/usecase/RegistrationTask.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.me.registrationtask.module.UserEntity
+import com.me.registrationtask.repository.IUserRepository
+
+//THIS CLASS CONSTRUCTOR TAKES IN A PARAMETER FROM THE IUserRepository AS AN OBJECT
+class RegistrationTask constructor(private val userRepository: IUserRepository) {
+
+    fun executeTask(userEntity: UserEntity) {
+
+        if (userEntity.Username.isEmpty()) throw IllegalArgumentException("Username is empty...")
```
Comment (alvinmarshall) [inline app/src/main/java/com/me/registrationtask/usecase/RegistrationTask.kt:11]: xup with the "..." ?

**Review by alvinmarshall — COMMENTED** [review]
Anyway good job 👍

**Review by alvinmarshall — APPROVED** [review]

**Pull request closed without merging**